libsrc/vic20/tgi/vic20-hi.s

```diff
@@ -0,0 +1,580 @@
+        .byte $00
+        .byte $0c
+
+XADDRS_L:
+        .byte <(VBASE + YRES * 0)
+        .byte <(VBASE + YRES * 1)
+        .byte <(VBASE + YRES * 2)
+        .byte <(VBASE + YRES * 3)
+        .byte <(VBASE + YRES * 4)
+        .byte <(VBASE + YRES * 5)
+        .byte <(VBASE + YRES * 6)
+        .byte <(VBASE + YRES * 7)
+        .byte <(VBASE + YRES * 8)
+        .byte <(VBASE + YRES * 9)
+        .byte <(VBASE + YRES * 10)
+        .byte <(VBASE + YRES * 11)
+        .byte <(VBASE + YRES * 12)
+        .byte <(VBASE + YRES * 13)
+        .byte <(VBASE + YRES * 14)
+        .byte <(VBASE + YRES * 15)
+        .byte <(VBASE + YRES * 16)
+        .byte <(VBASE + YRES * 17)
+        .byte <(VBASE + YRES * 18)
+        .byte <(VBASE + YRES * 19)
+
+XADDRS_H:
+        .byte >(VBASE + YRES * 0)
+        .byte >(VBASE + YRES * 1)
+        .byte >(VBASE + YRES * 2)
+        .byte >(VBASE + YRES * 3)
+        .byte >(VBASE + YRES * 4)
+        .byte >(VBASE + YRES * 5)
+        .byte >(VBASE + YRES * 6)
+        .byte >(VBASE + YRES * 7)
+        .byte >(VBASE + YRES * 8)
+        .byte >(VBASE + YRES * 9)
+        .byte >(VBASE + YRES * 10)
+        .byte >(VBASE + YRES * 11)
+        .byte >(VBASE + YRES * 12)
+        .byte >(VBASE + YRES * 13)
+        .byte >(VBASE + YRES * 14)
+        .byte >(VBASE + YRES * 15)
+        .byte >(VBASE + YRES * 16)
+        .byte >(VBASE + YRES * 17)
+        .byte >(VBASE + YRES * 18)
+        .byte >(VBASE + YRES * 19)
+
+.code
+
+; ------------------------------------------------------------------------
+; INSTALL routine. Is called after the driver is loaded into memory. May
+; initialize anything that has to be done just once. Is probably empty
+; most of the time.
+;
+; Must set an error code: NO
+;
+
+INSTALL:
+        rts
+
+
+; ------------------------------------------------------------------------
+; UNINSTALL routine. Is called before the driver is removed from memory. May
+; clean up anything done by INSTALL but is probably empty most of the time.
+;
+; Must set an error code: NO
+;
+
+UNINSTALL:
+        rts
+
+
+; ------------------------------------------------------------------------
+; INIT: Changes an already installed device from text mode to graphics
+; mode.
+; Note that INIT/DONE may be called multiple times while the driver
+; is loaded, while INSTALL is only called once, so any code that is needed
+; to initializes variables and so on must go here. Setting palette and
+; clearing the screen is not needed because this is called by the graphics
+; kernel later.
+; The graphics kernel will never call INIT when a graphics mode is already
+; active, so there is no need to protect against that.
+;
+; Must set an error code: YES
+;
+
+INIT:
+
+; Initialize variables
+
+        ldx     #$FF
+        stx     BITMASK
+
+; Make screen columns.
+        lda     #<SBASE
+        sta     tmp2
+        lda     #>SBASE
+        sta     tmp2+1
+        ldx     #0
+l:      ldy     #0
```

Comment:
Please use either:
- Cheap local labels
- Local labels named `L<number>` (as used in the code below)
- Local labels with "speaking" names
- "Normal" labels with "speaking" names